docs/developer/using-makefiles.md

```diff
@@ -0,0 +1,53 @@
+# How to set up the environment using the provided makefiles
+
+The main file is [Makefile](https://github.com/kubeapps/kubeapps/blob/master/Makefile), which will compile and prepare the production assets for then generating a set of Docker images. It is the starting point when you want to build the Kubeapps different components.
+
+For setting up the environment for running Kubeapps, we also provide (as it) makefiles for:
+* Creating a multicluster environment with Kind ([cluster-kind.mk](https://github.com/kubeapps/kubeapps/blob/master/script/cluster-kind.mk))
+* Deploying and configuring the components for getting Kubeapps running with OIDC login using Dex ([deploy-dev.mk](https://github.com/kubeapps/kubeapps/blob/master/script/deploy-dev.mk)).
+
+## Makefile for generating images
+
+### Commands
+
+Find below a list of the most used commands:
+
+``` bash
+make # will make all the kubeapps images
+make kubeapps/dashboard
+make kubeapps/apprepository-controller
+make kubeapps/kubeops
+make kubeapps/assetsvc
+make kubeapps/asset-syncer
+```
+
+> You can set the image tag manually: `IMAGE_TAG=myTag make`
+
+## Makefile for setting up the environment
+
+### Prerequisites
+
+* Install `mkcert`; you gan get it from the [official repository](https://github.com/FiloSottile/mkcert/releases).
+* Get the Kind network IP and replace it when necessary.
+    * Retrieve the IP by executing  `kubectl --namespace=kube-system get pods -o wide | grep kube-apiserver-kubeapps-control-plane  | awk '{print $6}'`
```

Comment:
Unfortunately, I don't think this will work, unless you mean that afterwards you `make delete-cluster-kind && make cluster-kind`, as some of these files (the apiserver-config ones) are config for the cluster api server itself, which has to know where to find dex.

I haven't found a way to provide an IP address for dex which is available to:

1. the api server and oauth2-proxy within the cluster,
2. the browser outside of the cluster

other than using a bridge network IP. I used to instead rely on setting an `/etc/hosts` line for `dex.local` but had to manually update the `/etc/hosts` of the api server (that's the orphaned .PHONY target you found, `update-apiserver-etc-hosts`) which was fiddly as it's not persistent (and may not even work now with distroless). Let me know if you've ideas, as I suspect we'll continue to have this issue with `tilt` or bash scripts.

Reply:
Mmm, that's true, you're right. Since I was constantly creating and deleting the kind cluster, it indeed worked for me.
I don't see at the moment a way to sort out this issue rather than yours. 
I'm going to update the docs with this information and let's wait until we come up with a better environment setup with tilt.